loader/proto.go

```diff
@@ -751,6 +723,454 @@ func (b *builder) handlePackage() (string, error) {
+
+	for i := 0; i < s.NumField(); i++ {
+		name := s.Field(i).Name
+		value := v.FieldByName(name)
+		if !isNilOrEmpty(value.Interface()) {
+			k := value.Kind()
+			switch {
+			case k >= reflect.Int && k <= reflect.Int64,
+				k == reflect.Float32 || k == reflect.Float64,
+				k >= reflect.Uint && k <= reflect.Uint64,
+				k == reflect.String,
+				k == reflect.Bool:
+				// The only way to know if the value
+				// is of type alias is to check its type.
+				if isGoType(value.Type().String()) {
+					b.format(w, 0, nil, "// %s: %v,\n", name, value)
+				} else {
+					pkg := strings.Split(reflect.TypeOf(value.Interface()).String(), ".")[0]
+					b.format(w, 0, nil, "// %s: %s.%v,\n", name, pkg, value)
+				}
+			case k == reflect.Map:
+				mapKey := value.Type().Key()
+				mapValue := strings.Replace(value.Type().Elem().String(), "*", "", -1)
+				b.format(w, 0, nil, "// %s: map[%s]%s{\n", name, mapKey, mapValue)
+				for _, key := range value.MapKeys() {
+					c := value.MapIndex(key)
+					switch reflect.TypeOf(c.Interface()).Kind() {
+					case reflect.Ptr:
+						t := reflect.Indirect(c).Interface()
+						b.format(w, 0, nil, "// %s: %s{\n", key, reflect.TypeOf(t))
+						b.format(w, 0, nil, b.fromStructToAnnotation(t))
+						b.format(w, 0, nil, "// },\n")
+					default:
+						b.format(w, 0, nil, "// %s: %s,\n", key, c)
+					}
+				}
+				b.format(w, 0, nil, "// },\n")
+			case k == reflect.Ptr:
+				t := reflect.Indirect(value).Interface()
+				b.format(w, 0, nil, "// %s: %s{\n", name, reflect.TypeOf(t))
+				w.WriteString(b.fromStructToAnnotation(t))
+				b.format(w, 0, nil, "// },\n")
+			case k == reflect.Slice:
+				b.format(w, 0, nil, "// %s: %s{\n", name, reflect.TypeOf(value.Interface()))
+				for i := 0; i < value.Len(); i++ {
+					val := value.Index(i).Interface()
+					switch reflect.TypeOf(val).Kind() {
+					case reflect.Ptr:
+						// TODO (ced)
+					default:
+						if isGoType(reflect.TypeOf(val).String()) {
+							b.format(w, 0, nil, "// %v,\n", val)
+						} else {
+							pkg := strings.Split(reflect.TypeOf(val).String(), ".")[0]
+							b.format(w, 0, nil, "// %s.%v,\n", pkg, val)
+						}
+					}
+				}
+				b.format(w, 0, nil, "// },\n")
+			}
+		}
+	}
+	return w.String()
+}
+
+func isNilOrEmpty(x interface{}) bool {
+	return reflect.DeepEqual(x, reflect.Zero(reflect.TypeOf(x)).Interface())
+}
+
+func (b *builder) convertSwagger(lit proto.Literal) (*openapiv2.Swagger, error) {
+	if len(lit.OrderedMap) == 0 {
+		return nil, fmt.Errorf("expected swagger option to be a map")
+	}
+	swagger := &openapiv2.Swagger{
+		Responses: map[string]*openapiv2.Response{},
+	}
+	for _, v := range lit.OrderedMap {
+		switch v.Name {
+		case "responses":
+			key, resp, err := b.convertResponse(v)
+			if err != nil {
+				return nil, err
+			}
+			swagger.Responses[key] = resp
+		case "swagger":
+			swagger.Swagger = v.SourceRepresentation()
+		case "base_path":
+			swagger.BasePath = v.SourceRepresentation()
```

Comment:
I can't help but find this too repetitive. Any thoughts on automating this more?

Reply:
Indeed, I'll do it using reflection